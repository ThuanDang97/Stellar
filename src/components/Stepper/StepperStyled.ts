import styled from 'styled-components'

// Utils
import { pxToRem } from '@utils/theme'

// Components
import { TextStyled } from '@components/styled-components'

export const ArrowStyled = styled(TextStyled)`
  border-top: ${({ theme }) => pxToRem(theme.metrics.borderWidth.base)} solid
    transparent;
  border-bottom: ${({ theme }) => pxToRem(theme.metrics.borderWidth.base)} solid
    transparent;
  border-left: ${({ theme }) => pxToRem(theme.metrics.borderWidth.lg)} solid
    ${({ theme }) => theme.colors.pomegranate};
  margin-left: ${({ theme }) => pxToRem(theme.metrics.dimensions.sm)};
  margin-right: ${({ theme }) => pxToRem(theme.metrics.dimensions.sm)};
`
