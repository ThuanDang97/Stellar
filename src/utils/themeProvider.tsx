import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
