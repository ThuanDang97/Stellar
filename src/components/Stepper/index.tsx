import { FC, memo } from 'react'

// Props type
import { StepperProps } from '@self-types/components/Stepper.props'

// Themes
import { theme } from '@themes/index'

// Constants
import { STEPPER } from '@constants/index'

// Components
import { TextStyled, ViewStyled } from '@components/styled-components'
import { ArrowStyled } from './StepperStyled'

const Stepper: FC<StepperProps> = ({ currentPage = STEPPER[0].step }) => (
  <ViewStyled display="flex">
    {STEPPER.map((item) => (
      <ViewStyled display="flex" key={item.step}>
        <TextStyled
          fontWeight={theme.typography.fontWeight.base}
          color={
            currentPage === item.id ? theme.colors.cello : theme.colors.silver
          }
          data-testid="stepper-value"
        >
          {item.step}
        </TextStyled>
        {STEPPER[STEPPER.length - 1].step !== item.step && <ArrowStyled />}
      </ViewStyled>
    ))}
  </ViewStyled>
)

export default memo(Stepper)
