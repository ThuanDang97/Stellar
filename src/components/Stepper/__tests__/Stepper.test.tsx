import { screen } from '@testing-library/react'

// Components
import Stepper from '@components/Stepper'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

// Constants
import { STEPPER } from '@constants/index'

describe('Stepper Component', () => {
  const component = <Stepper />

  test('Component Stepper matchers DOM Snapshot', () => {
    const { container } = renderWithTheme(component)

    expect(container).toMatchSnapshot()
  })

  test('Component Stepper should render default correctly', () => {
    renderWithTheme(component)

    const items = screen.getAllByTestId('stepper-value')

    expect(items[0]).toHaveTextContent(STEPPER[0].step)
  })
})
