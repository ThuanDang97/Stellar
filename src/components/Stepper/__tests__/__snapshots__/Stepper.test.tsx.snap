// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper Component Component Stepper matchers DOM Snapshot 1`] = `
<div>
  <div
    class="ViewStyled-sc-rqe5sa-0 knVWgS"
    display="flex"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 knVWgS"
      display="flex"
    >
      <span
        class="TextStyled-sc-9xia41-0 leTdwL"
        color="#C0C0C0"
        data-testid="stepper-value"
        font-weight="400"
      >
        Sign Up
      </span>
      <span
        class="TextStyled-sc-9xia41-0 StepperStyled__ArrowStyled-sc-1rhowde-0 gtHdGO jbeCTy"
      />
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 knVWgS"
      display="flex"
    >
      <span
        class="TextStyled-sc-9xia41-0 leTdwL"
        color="#C0C0C0"
        data-testid="stepper-value"
        font-weight="400"
      >
        Set Goals
      </span>
      <span
        class="TextStyled-sc-9xia41-0 StepperStyled__ArrowStyled-sc-1rhowde-0 gtHdGO jbeCTy"
      />
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 knVWgS"
      display="flex"
    >
      <span
        class="TextStyled-sc-9xia41-0 leTdwL"
        color="#C0C0C0"
        data-testid="stepper-value"
        font-weight="400"
      >
        Connect Bank
      </span>
    </div>
  </div>
</div>
`;
